### ISSUE TITLE
border short-hand breaks when using rgba(…) color

### ISSUE THREAD

**Issue by blixt** [jakubpawlowicz/clean-css#638]
This:

``` css
.viewport .download a {
    border: rgba(255, 255, 255, .6) solid 1px;
    border-radius: 9999px;
    color: #fff;
    display: inline-block;
    font-size: 18px;
    padding: 10px 20px;
    text-decoration: none;
}
```

Gets turned into this:

``` css
.viewport .download a{border:rgba(255,255,255,.6) solid;border-radius:9999px;color:#fff;display:inline-block;font-size:18px;padding:10px 20px;text-decoration:none}
```

Note how the `1px` was dropped from the `border` short-hand. This causes the border to have the wrong width.

**Comment by jakubpawlowicz** [comment]
Apparently we miss fuzzy matching for border properties, which, according to spec, come in <width> <color> <style> order. It's not an excuse as browsers support a mixed order too.

**Comment by davidhund** [comment]
I just noticed the same bug with `outline` shorthand:

`* { outline: 1px dotted rgb(255,0,0); }` → `*{outline:red dotted 1px}`
`* { outline: 1px dotted rgba(255,0,0,.5); }` → `*{outline:dotted rgba(255,0,0,.5)}`

…The 1px is dropped resulting in *initial* `outline-width` of ~3px :fire: 

**However:** it's not a problem with `rgba()` alone because an opacity value of 1 works:

`* { outline: 1px dotted rgba(255,0,0,1); }` → `*{outline:red dotted 1px}`

:ok_hand: `rgba(0,0,0, 1)`
:-1: `rgba(0,0,0, .9)` (outline-width is dropped)

My guess is that it's the same with the `border` shorthand.

**Comment by davidhund** [comment]
**PPS: It get's weirder**

I just discovered the universal star `*` selector plays a part, as does `:before` and `:after`
Using these together with `rgba()` (with opacity < 1) causes the bug that drops the `outline-width`.

:-1: `* { outline: 1px dotted rgba(0,0,0, .5); }`
:-1: `:before { outline: 1px dotted rgba(0,0,0, .5); }`
:-1: `:after { outline: 1px dotted rgba(0,0,0, .5); }`

…combine these with another selector and they work:
:+1: `body * { outline: 1px dotted rgba(0,0,0, .5); }`
:+1: `.foo:before { outline: 1px dotted rgba(0,0,0, .5); }`
:+1: `bar:after { outline: 1px dotted rgba(0,0,0, .5); }`

The order of `rgba()` does not seem to matter with outline, by the way.

Should I create a different bug for outline?

**Comment by jakubpawlowicz** [comment]
@davidhund hmmm, that's odd as I can neither replicate it with 3.3.3 or 3.4.1, e.g.: 

```bash
$> echo "* { outline: 1px dotted rgba(0,0,0, .5); }" | cleancss
*{outline:rgba(0,0,0,.5) dotted 1px}
```

Am I doing it correctly?
It looks like a separate issue, so please open a new one.

**Comment by davidhund** [comment]
Weird. I get that too using either `cleancss` directly or using the `gulp-minify-css` version. Will look into it some more and open a separate bug.

**Comment by jakubpawlowicz** [comment]
Cool, thanks!

**Comment by jakubpawlowicz** [comment]
Sorry for the delay.

This bug has been fixed on 3.4 for quite a while. It works OK on master too.
Tests are already in place since d72356a5.
